googleapiclient/discovery_cache/__init__.py

```diff
@@ -47,3 +48,36 @@ def autodetect():
     except Exception as e:
         LOGGER.warning(e, exc_info=True)
         return None
+
+def get_static_doc(uri):
+    """Retrieves the discovery document from the directory defined in
+    DISCOVERY_DOC_STATIC_DIR corresponding to the uri provided.
+
+    Args:
+        uri: string, The URI of the discovery document in the format
+        https://{domain}/discovery/{discoveryVer}/apis/{api}/{apiVersion}/rest
+
+    Returns:
+        A string containing the contents of the JSON discovery document,
+        otherwise None if the JSON discovery document was not found.
+    """
+
+    doc_name = None
+    content = None
+
+    # Extract the {apiVersion} and {api} from the uri which are the 2nd and 3rd
+    # last parts of the uri respectively.
+    # https://www.googleapis.com/discovery/v1/apis/{api}/{apiVersion}/rest
```

Comment:
There are two different URI patterns (see `V1_DISCOVERY_URI` and `V2_DISCOVERY_URI` in `discovery.py`). Perhaps the service name and version can be passed from `build()` to `_retrieve_discovery_doc()` so the uri doesn't have to be parsed here?

Reply:
Good catch! Fixed.